reign.py

```diff
@@ -116,15 +121,27 @@ def battle(self):
         """
 
         # Choose ∑
-        empire = random.choice(self.obj.Empire.values.tolist())
-        empire_neighbours = self.obj.query(f'Empire == "{empire}"').iloc[0].empire_neighbours
+
+        empire_list = self.obj.Empire.values.tolist()
+        un_empire, empire_weight = np.unique(empire_list, return_counts=True)
+        probability_ratio = max(empire_weight)/empire_weight
+
+        if len(un_empire) < self.threshold and any(probability_ratio > self.low_b):
```

Comment:
This line needs a comment.
Does `len(un_empire)` means the size of the empire?
Why do we use `any(probability_ratio > self.low_b)`?

Reply:
`len(un_empire)` is the amount of empires in play. 
`any(probability_ratio > self.low_b)` returns True if at least 1 of the empire weight exceed the lower boundary. To be precise the lower boundary is set as being 1/3 of the highest weight.